### PR TITLE
Simplify the API for launching Flutter

### PR DESCRIPTION
**Pull request by stuartmorgan**
- Eliminate the distinction between source mode and snapshot mode;
  source mode hasn't worked since the switch to Dart2, and the relevant
  engine arguments are deprecated in
  https://github.com/flutter/engine/pull/7497
- Hide the need for the first argument to be the executable from clients
  of the library; this is an implementation detail of the engine API,
  and not something that each client application should need to handle
  given that the expected use case is to provide a constructed argument
  array, rather than pass through argc/argv directly.
  (As part of this change, removes the macOS code to strip Xcode-added
  arguments, for the same reason.)

This is a breaking change; clients will need to update the call to the
primary entry point, including removing the dummy first argument they
were likely providing in the arguments array.

### CODE REVIEW

File: library/common/glfw/embedder.cc [hunk 1]
```diff
@@ -246,11 +246,13 @@ static void *GLFWProcResolver(void *user_data, const char *name) {
...
 // Returns a caller-owned pointer to the engine.
 static FlutterEngine RunFlutterEngine(
-    GLFWwindow *window, const std::string &main_path,
-    const std::string &assets_path, const std::string &packages_path,
+    GLFWwindow *window, const std::string &assets_path,
     const std::string &icu_data_path,
     const std::vector<std::string> &arguments) {
-  std::vector<const char *> argv;
+  // FlutterProjectArgs is expecting a full argv, so when processing it for
+  // flags the first item treated as the executable and ignored. Add a dummy
```
Comment (franciscojma86) [inline library/common/glfw/embedder.cc]: s/and/is ?

File: library/macos/FLEViewController.m [hunk 2]
```diff
@@ -280,38 +263,30 @@ - (BOOL)launchEngineInternalWithAssetsPath:(NSURL *)assets
...
-  for (int i = 0; i < argc; ++i) {
-    if ([arguments[i] isEqualToString:kXcodeExtraArgumentOne] && (i + 1 < argc) &&
-        [arguments[i + 1] isEqualToString:kXcodeExtraArgumentTwo]) {
-      unused += 2;
-      ++i;
-      continue;
-    }
-    command_line_args[i - unused] = [arguments[i] UTF8String];
+  // FlutterProjectArgs is expecting a full argv, so when processing it for flags the first
+  // item treated as the executable and ignored. Add a dummy value so that all provided arguments
```
Comment (franciscojma86) [inline library/macos/FLEViewController.m]: same here

**Comment by franciscojma86** [comment]
LGTM with with two typo comments

File: library/common/glfw/embedder.cc (hunk 1 above)
Comment (stuartmorgan) [reply library/common/glfw/embedder.cc]: Good catch; should have been "the first item **is** treated as the executable and ignored". Fixed in both files.

**Pull request merged**